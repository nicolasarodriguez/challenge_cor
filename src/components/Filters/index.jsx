import React from 'react'
import { func } from 'prop-types'

import { STATUS, PRIORITY } from '../../utils/constants'
import { createDropDown } from '../../utils/utils'

const namespace = 'ui-filters'

const Filters = ({ setPriorityFilter, setStatusFilter }) => {
  return (
    <div className={`${namespace}`}>
      <h3 className={`${namespace}__title`}>Filtar por:</h3>
      <div className={`${namespace}__box`}>
        <div>
          <span>Prioridad</span>
          <select onChange={e => setPriorityFilter(e.target.value)}>
              <option value='all'>Todos</option>
              { createDropDown(PRIORITY) }
          </select>
        </div>
        <div>
          <span>Estado</span>
          <select onChange={e => setStatusFilter(e.target.value)}>
              <option value='all'>Todos</option>
              { createDropDown(STATUS) }
          </select>
        </div>
      </div>
    </div>
  )
}

Filters.propTypes = {
  setPriorityFilter: func.isRequired,
  setStatusFilter: func.isRequired
}

export default Filters
