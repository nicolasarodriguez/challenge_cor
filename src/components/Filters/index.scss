@import '../../styles/variales.scss';

.ui-filters {
    margin: 0 auto 20px;
    max-width: $max-width;

    &__title {
        margin: 0 0 10px;
    }

    &__box {
        display: flex;
        grid-gap: 20px;

        span {
            padding: 0 10px 0 0;
        }
    }
}

@media (max-width: $breakpoint) {
    .ui-filters {
        &__box {
            display: block;

            div {
                margin-bottom: 10px;
            }
        }
    }
}
