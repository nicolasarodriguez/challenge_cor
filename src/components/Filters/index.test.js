import React from 'react';
import { render, screen } from '@testing-library/react';

import Filters from './index';

const defaultProps = {
  setStatusFilter: jest.fn(),
  setPriorityFilter: jest.fn(),
}

const setup = (propOverrides = {}) => {
    const props = {
      ...defaultProps,
      ...propOverrides,
    };
  
    const view = render(<Filters {...props} />);
  
    return {
      props,
      view
    };
};

describe('<Filters />', () => {
    it('should match snapshot', () => {
        const { view } = setup();
        const Button = screen.getByText('Filtar por:');
        expect(Button).toBeInTheDocument();
        expect(view).toMatchSnapshot();
    });
});
