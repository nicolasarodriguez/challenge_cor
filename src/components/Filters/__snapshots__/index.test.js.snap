// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui-filters"
      >
        <h3
          class="ui-filters__title"
        >
          Filtar por:
        </h3>
        <div
          class="ui-filters__box"
        >
          <div>
            <span>
              Prioridad
            </span>
            <select>
              <option
                value="all"
              >
                Todos
              </option>
              <option
                value="HIGH"
              >
                Alta
              </option>
              <option
                value="MEDIUM"
              >
                Media
              </option>
              <option
                value="LOW"
              >
                Baja
              </option>
            </select>
          </div>
          <div>
            <span>
              Estado
            </span>
            <select>
              <option
                value="all"
              >
                Todos
              </option>
              <option
                value="NEW"
              >
                Nueva
              </option>
              <option
                value="IN_PROGRESS"
              >
                En proceso
              </option>
              <option
                value="DONE"
              >
                Finalizada
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui-filters"
    >
      <h3
        class="ui-filters__title"
      >
        Filtar por:
      </h3>
      <div
        class="ui-filters__box"
      >
        <div>
          <span>
            Prioridad
          </span>
          <select>
            <option
              value="all"
            >
              Todos
            </option>
            <option
              value="HIGH"
            >
              Alta
            </option>
            <option
              value="MEDIUM"
            >
              Media
            </option>
            <option
              value="LOW"
            >
              Baja
            </option>
          </select>
        </div>
        <div>
          <span>
            Estado
          </span>
          <select>
            <option
              value="all"
            >
              Todos
            </option>
            <option
              value="NEW"
            >
              Nueva
            </option>
            <option
              value="IN_PROGRESS"
            >
              En proceso
            </option>
            <option
              value="DONE"
            >
              Finalizada
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
